tests/pyright_test.py

```diff
@@ -29,12 +29,15 @@ def main() -> None:
         print("error running npx; is Node.js installed?", file=sys.stderr)
         sys.exit(1)
 
-    with open("pyproject.toml", "rb") as config:
-        pyright_version: str = tomli.load(config)["tool"]["typeshed"]["pyright_version"]
+    req = parse_requirements()["pyright"]
+    assert len(req.specifier) == 1
+    spec = str(req.specifier)
+    assert spec.startswith("=="), f"pyright version must be pinned in requirements-tests.txt: {req}"
```

Comment:
Shoudldn't we be asserting these invariants in this function, along with all the other type checker pins, rather than here? https://github.com/python/typeshed/blob/35818460691b7d1431a4ccc9524339ff38da48f3/tests/check_consistent.py#L140-L148

You should just need to add `"pyright"` to this set here: https://github.com/python/typeshed/blob/35818460691b7d1431a4ccc9524339ff38da48f3/tests/check_consistent.py#L28